hack/verify-config.sh

```diff
@@ -23,13 +23,12 @@ TESTINFRA_ROOT=$(git rev-parse --show-toplevel)
 PROW_CONFIG="${TESTINFRA_ROOT}/prow/config.yaml"
 JOBS_CONFIG="${TESTINFRA_ROOT}/jobs/config.json"
 TMP_CONFIG=$(mktemp)
-trap "rm $TMP_CONFIG" EXIT
+trap 'rm "$TMP_CONFIG"' EXIT
 cp "${PROW_CONFIG}" "${TMP_CONFIG}"
-bazel run //maintenance/fixconfig:fixconfig -- --config=${TMP_CONFIG} --config-json=${JOBS_CONFIG} && \
-bazel run //jobs:config_sort -- --prow-config=${TMP_CONFIG} --only-prow
+bazel run //maintenance/fixconfig:fixconfig -- --config="${TMP_CONFIG}" --config-json="${JOBS_CONFIG}" && \
+bazel run //jobs:config_sort -- --prow-config="${TMP_CONFIG}" --only-prow
 
-DIFF=$(diff "${TMP_CONFIG}" "${PROW_CONFIG}")
-if [ ! -z "$DIFF"]; then
-    echo "config is not correct, please run `hack/update-config.sh`"
+diff -q "${TMP_CONFIG}" "${PROW_CONFIG}" || (
```

Comment:
Why? This is pretty non-standard in my experience and people who don't know shell that well will 10/10 want `if ... then` rather than this.